### PR TITLE
fix azure GetSecretMap

### LINKED ISSUE #263: AzureKV dataFrom implementation returns all secrets
The azure `dataFrom` implementation returns all `secrets` of the vault while the `ref.Key` is not used at all. 

It should:
* respect the well-known prefixes to select an object-type: `secret|cert|key`
* return an error when specifying an `cert` or `key` 
* a secret value can be interpreted as json-string so it can be used with `ref.Property`

https://github.com/external-secrets/external-secrets/blob/0613b904b620d5332c38ab3ba7c0aa4e740dd2c5/pkg/provider/azure/keyvault/keyvault.go#L139-L148

### PR DESCRIPTION
**Pull request by Jabray5**
Change Azure GetSecretMap to be similar to GCP implementation. Unit tests are not working for GetSecretMap - to be fixed in a future pull request. Closes #263

### CODE REVIEW

File: pkg/provider/azure/keyvault/keyvault.go [hunk 1]
```diff
@@ -142,37 +141,25 @@ func (a *Azure) GetSecret(ctx context.Context, ref esv1alpha1.ExternalSecretData
...
-		}
-		err = secretListIter.Next()
-		if err != nil {
-			return nil, err
-		}
+
+	kv := make(map[string]string)
+	err = json.Unmarshal(data, &kv)
+	if err != nil {
+		return nil, fmt.Errorf("Error unmarshalling json data: %w", err)
```
Comment (knelasevero) [inline pkg/provider/azure/keyvault/keyvault.go:154]: Hey @Jabray5, So this first letter of the error string should not be capitalized :)

**Comment by knelasevero** [comment]
Oh, also we should return an error if they use other types like cert and key.

**Comment by knelasevero** [comment]
I think this LGTM, but since I helped Joey with this one, wait for another set of eyes

**Comment by moolen** [comment]
Looks good to me, thank you for the fix :tada:! reminder: for the release notes we should add a note at the top to inform users that this particular functionality has changed.

**Pull request merged**